server/routes/index.js

```diff
@@ -1,7 +1,17 @@
 const router = require('express').Router();
 
+const {signup} = require('../controllers');
+const { getProductByCId } = require('../controllers/getProductsByCId');
 const { getProducts, getProductData } = require('../controllers/products');
 
+
+router.post('/register', signup);
+
+router.get('categories/:productId', getProductByCId)
```

Comment:
Shouldn't this be `categoryId`?
```suggestion
router.get('categories/:categoryId', getProductByCId)
```